digits/model/images/job.py

```diff
@@ -17,3 +21,65 @@ def __init__(self, **kwargs):
         super(ImageModelJob, self).__init__(**kwargs)
         self.pickver_job_model_image = PICKLE_VERSION
 
+    def get_snapshot_filename(self,epoch=-1):
+        task = self.train_task()
+        snapshot_filename = None
+        if epoch == -1 and len(task.snapshots):
+            epoch = task.snapshots[-1][1]
+            snapshot_filename = task.snapshots[-1][0]
+        else:
+            for f, e in task.snapshots:
+                if e == epoch:
+                    snapshot_filename = f
+                    break
+        if not snapshot_filename:
+            raise ValueError('Invalid epoch')
+
+        path, name = os.path.split(snapshot_filename)
+
+        return name
+
+    @override
+    def get_job_stats_as_json_string(self,epoch=-1):
+        task = self.train_task()
+
+        loc, mean_file = os.path.split(task.dataset.get_mean_file())
+
+        stats = {
+            "job id": self.id(),
+            "creation time": self.status_history[0][1],
+            "username": self.username,
+            "image dimensions": task.dataset.get_feature_dims(),
+            "mean file": mean_file,
+            "snapshot file": self.get_snapshot_filename()
+        }
+
+        # Set the torch and caffe specific attributes:
+        if task.name() == 'Train Torch Model':
```

Comment:
I would move this to the train task. With proper class inheritance you will be able to do Torch-specific stuff in `torch_train.py` and Caffe-specific stuff to `caffe_train.py`. As a rule of thumb, never write framework-specific code outside of framework-specific files - doing this breaks framework agnosticity.